Makefile

```diff
@@ -16,105 +16,211 @@
 # under the License.
 
 SHELL := /bin/bash
+
 # We assume an active virtualenv for development
-VIRTUAL_ENV ?= .venv
+VIRTUAL_ENV := $(or $(VIRTUAL_ENV),.venv$(if $(PY_VERSION),-$(PY_VERSION)))
 VENV_ACTIVATE_FILE := $(VIRTUAL_ENV)/bin/activate
-VE_MISSING_HELP := "\033[0;31mIMPORTANT\033[0m: Couldn't find $(PWD)/$(VIRTUAL_ENV); have you executed make install?\033[0m\n"
+VENV_ACTIVATE := source $(VENV_ACTIVATE_FILE)
 
 PY_VERSION := $(shell jq -r '.python_versions.DEFAULT_PY_VER' .ci/variables.json)
+export UV_PYTHON := $(PY_VERSION)
+export UV_PROJECT_ENVIRONMENT := $(VIRTUAL_ENV)
 
-.PHONY: install \
-	check-venv \
-	reinstall \
-	check-uv \
-	venv-destroy \
+PRE_COMMIT_HOOK_PATH := .git/hooks/pre-commit
+
+LOG_CI_LEVEL := INFO
+
+# --- Global goals ---
+
+all: lint test it
+
+clean: clean-others clean-docs
+
+check-all: all
+
+.PHONY: \
+	all \
 	clean \
+	check-all \
+	uv \
 	uv-add \
 	uv-lock \
-	nondocs-clean \
-	docs-clean \
-	python-caches-clean \
+	venv \
+	clean-venv \
+	install_pytest_rally_plugin \
+	install \
+	reinstall \
+	venv-destroy \
+	clean-others \
+	clean-pycache \
 	lint \
 	format \
+	pre-commit \
+	install-pre-commit \
 	docs \
 	serve-docs \
+	clean-docs \
 	test \
+	test-all \
+	test-3.10 \
+	test-3.11 \
+	test-3.12 \
+	test-3.13 \
 	it \
-	check-all \
+	it_serverless \
+	it_tracks_compat \
 	benchmark \
 	release-checks \
-	release
+	release \
+	sh
 
-install: check-uv
-	uv sync --python $(PY_VERSION) --locked --extra=develop
+# --- uv goals ---
 
-reinstall: venv-destroy install
-
-check-venv:
-	@if [[ ! -f $(VENV_ACTIVATE_FILE) ]]; then \
-		printf $(VE_MISSING_HELP); \
-	fi
-
-check-uv:
+# It checks uv is installed.
+uv:
 	@if [[ ! -x $$(command -v uv) ]]; then \
-		printf "Please install uv by running the following outside of a virtual env: [ curl -LsSf https://astral.sh/uv/install.sh | sh ]\n"; \
+		printf "Please install uv by running the following outside of a virtual environment (https://docs.astral.sh/uv/getting-started/installation/)"; \
 	fi
 
-venv-destroy:
-	@echo "Removing virtual environment $(VIRTUAL_ENV)"
-	rm -rf $(VIRTUAL_ENV)
-
-clean: nondocs-clean docs-clean
-
+# It adds a list of packages to the project.
 uv-add:
 ifndef ARGS
 	$(error Missing arguments. Use make uv-add ARGS="...")
 endif
-	uv add --python $(PY_VERSION) $$ARGS
+	uv add $$ARGS
 
+# It updates the uv lock file.
 uv-lock:
-	uv lock --python $(PY_VERSION)
+	uv lock
 
-nondocs-clean:
-	rm -rf .benchmarks .eggs .nox .rally_it .cache build dist esrally.egg-info logs junit-py*.xml NOTICE.txt
 
-docs-clean:
-	cd docs && $(MAKE) clean
+# --- venv goals ---
+
+# It creates the project virtual environment.
+venv: uv $(VENV_ACTIVATE_FILE)
+
+$(VENV_ACTIVATE_FILE):
+	uv venv --allow-existing --seed
+	uv sync --locked --extra=develop
```

Comment:
Any particular reason for reintroducing `pip` with `--seed`? Isn't the 2nd command enough?